src/storage-client/src/source/persist_source.rs

```diff
@@ -74,7 +75,7 @@ where
     G: Scope<Timestamp = mz_repr::Timestamp>,
     YFn: Fn(Instant, usize) -> bool + 'static,
 {
-    let (stream, token) = persist_source_core(
+    let (updates, errs, token) = persist_source_core(
```

Comment:
am I missing something or are persist_source and persist_source_core the same after this?

Reply:
Yep, it seems they are! I can change it as part of this PR, or not, whatever you prefer.

Reply:
if we're keeping the change to persist_source_core's return type, I lean toward yes, clean them up while we're in here making changes